### ISSUE TITLE
Jitify version is not consistent in different installation methods

### ISSUE THREAD

**Issue by leofang** [cupy/cupy#8153]
> So now we have two issues:
> 
> 1. With wheels (I confirmed on Linux too): Jitify version is unknown
> 2. With conda packages (on Linux): Jitify version is **not** unknown

Update: I just checked the Windows conda package and wheel. Below is an updated summary:
1. With wheels:
   a. On Linux: Jitify version is `<unknown>`
   b. On Windows: Jitify version is **not** `<unknown>`
2. With conda packages:
   a. On Linux: Jitify version is **not** `<unknown>`
   b. On Windows: Jitify version is `<unknown>`

@kmaehashi I am confused why the situation is flipped between wheel/conda and between Linux/Windows...

_Originally posted by @leofang in https://github.com/cupy/cupy/issues/8147#issuecomment-1924130611_

**Comment by leofang** [comment]
@kmaehashi any thoughts?

**Comment by kmaehashi** [comment]
> > 1. With wheels:
> >    a. On Linux: Jitify version is `<unknown>`
> >    b. On Windows: Jitify version is **not** `<unknown>`

Currently Windows wheels are built with `.git`, but Linux wheels are not. (Linux wheel build infra has recently been migrated to the new GitHub Actions-based pipeline and is currently in the private repo. I think this needs to be fixed...)

Not sure why Windows conda packages say `<unknown>`...

**Comment by leofang** [comment]
> Not sure why Windows conda packages say `<unknown>`...

We download sdist from PyPI to build conda packages, could this be relevant?

**Comment by kmaehashi** [comment]
> > Not sure why Windows conda packages say `<unknown>`...
> 
> We download sdist from PyPI to build conda packages, could this be relevant?

Likely. sdist does not contain `.git`.

**Comment by leofang** [comment]
> > > Not sure why Windows conda packages say `<unknown>`...
> > 
> > 
> > We download sdist from PyPI to build conda packages, could this be relevant?
> 
> Likely. sdist does not contain `.git`.

Indeed, it occurred to me that we rely only on `git` to give Jitify a version. But now I don't understand why the version in Linux conda packages is *not* `<unknown>`...
